public_html/wp-content/themes/pattern-directory/src/components/pattern-grid-menu/index.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import { store as coreStore } from '@wordpress/core-data';
+import { useSelect } from '@wordpress/data';
+import { useEffect, useState } from '@wordpress/element';
+import { getPath } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import CategoryMenu from '../category-menu';
+import CategorySearch from '../category-search';
+import CategoryContextBar from '../category-context-bar';
+import contextMessaging from './messaging';
+
+/**
+ * Module constants
+ */
+const PATTERN_TAXONOMY = 'wporg-pattern-category';
+
+const PatternGridMenu = () => {
+	// Show loading state
+	const [ showLoading, setShowLoading ] = useState( true );
+	const [ isFetching, setIsFetching ] = useState( false );
+	const [ path, setPath ] = useState();
+	const [ categoryContext, setCategoryContext ] = useState( undefined );
+
+	const categories = useSelect( ( select ) => select( coreStore ).getEntityRecords( 'taxonomy', PATTERN_TAXONOMY ) );
+
+	useEffect( () => {
+		const pathOnLoad = getPath( window.location.href );
+		setPath( pathOnLoad );
+	}, [] );
+
+	useEffect( () => {
+		// Since categories starts as an [] then switches to null
+		if ( showLoading && categories === null ) {
+			setIsFetching( true );
+		}
+
+		if ( isFetching && Array.isArray( categories ) ) {
+			setShowLoading( false );
+			setIsFetching( false );
+		}
+	}, [ isFetching, categories ] );
+
+	useEffect( () => {
+		setCategoryContext( contextMessaging[ path ] );
+	}, [ path ] );
+
+	return (
+		<div className="pattern-grid-menu">
+			<div className="pattern-grid-menu__actions">
+				<div>
```

Comment:
This is leading to a lot of nested divs, could any of these be `<>` fragments instead?

Reply:
I made some updates to remove some divs. It does make the component slightly less portable but it's a fair trade-off at this point.

Reply:
I don't think we have immediate plans to use this whole component anywhere else (well, maybe the `CategoryMenu`), so I wouldn't worry about making it reusable yet - we can always iterate if we need to.